### ISSUE TITLE
Jasypt ignores Custom StringEncryptor

### ISSUE THREAD

**Issue by DominikGeiger** [ulisesbocchio/jasypt-spring-boot#180]
I try to use the Jasypt Framework to decrypt properties which will be used by frameworks like flyway. 
Also I want to use my own DecryptorUtil which looks as follows:

```
public class MyCustomStringEncryptor implements org.jasypt.encryption.StringEncryptor {
    @Override
    public String encrypt(String message) {
        return MyEncryptorUtil.encrypt(message);
    }

    @Override
    public String decrypt(String encryptedMessage) {
        return MyEncryptorUtil.decrypt(encryptedMessage);
    }
}
```
this is implemented into my Spring Boot Application as follows:

```
@SpringBootApplication
@EnableEncryptableProperties
public class FFPortalApplication extends SpringBootServletInitializer {
	public static void main(String[] args) {
		final StandardEncryptableEnvironment environment = new StandardEncryptableEnvironment(new MyCustomStringEncryptor());
		
        new SpringApplicationBuilder()
            .environment(environment)
            .sources(FFPortalApplication.class)
            .run(args);
	}
}
```

Also note that my Configuration has the annotation 
`@EncryptablePropertySources({ @EncryptablePropertySource("classpath:application.properties") })`

Debugging into the StandardEncryptableEnvironment showed me, that the set encryptor is set into the environment but not for the DefaultPropertyResolver which leads to calling encryptor.decrypt(resolvedProperty); on the Default Encryptor and not the custom one. 

I found a workaround by extending StandardEncryptableEnvironment with MyCustomEncryptableEnvironment. 
```

public class MyCustomEncryptableEnvironment extends StandardEncryptableEnvironment {

    public MyCustomEncryptableEnvironment(InterceptionMode interceptionMode, StringEncryptor encryptor) {
        super(interceptionMode, encryptor);
        actuallyCustomizePropertySources();
    }
```

calling the actuallyCustomizePropertySources() method makes that the Custom String Encryptoris set when calling encryptor.decrypt(resolvedProperty). 
In my opinion if present the given encryptor must be set instead of the DefaultLazyEncryptor.

**Comment by ulisesbocchio** [comment]
should be fixed in 3.0.1

**Comment by DominikGeiger** [comment]
Works now. Thank you very much for the fast reply and fix!
